### DISCUSSION TITLE
Define document with two of the same output formats

### DISCUSSION THREAD

**Discussion by MikeLydeamore** [quarto-dev/quarto-cli#6480] [Q&A] [answered]
### Description

I am working on an extension to develop questions and solutions in a single output (https://github.com/MikeLydeamore/unilur). The idea is to produce to two HTML documents from the single QMD. These two formats share all of the same resources (the CSS and LUA filter), but I can't define two sets of HTML outputs in one extension.

I'd like to avoid having to duplicate the lua filter and have two extensions. Is there a way to share assets between extensions, or define multiple HTML output formats in a single extension?

**Comment by dragonstyle** [comment, marked as answer]
You could try using a `+` variant to differentiate the two. For example:

```yml
format:
  html:
    theme: flatly
  html+sketch:
    theme: sketchy
```

Users of your extension should then be able to do something like:

`quarto render index.qmd --to extensionname-html`

or

`quarto render index.qmd --to extensionname-html+sketchy`

I _think_ that should work...

Both formats would only be rendered if the user were to target them both, or use `--to all` or the like.

**Comment by MikeLydeamore** [comment]
Great this works a treat. I've implemented here: https://github.com/MikeLydeamore/unilur/commit/9d0ed24bb6ef14be8eba00dbb33cdae555dc8b36 https://github.com/MikeLydeamore/unilur/blob/main/_extensions/unilur/_extension.yml

Thanks for your help.

**Reply by dragonstyle** [reply]
Nice! It actually reads pretty well to add `+solution` to the format too!
